src/downloader/linux_updater.py

```diff
@@ -178,6 +186,51 @@ def _run_subprocesses(self, linux, linux_path):
             self._logger.print('ERROR! Something went wrong during the Linux update, try again later.')
             self._logger.print('Error code: %d' % result.returncode)
             self._logger.print()
+            return
+
+        if len(self._user_files) > 0:
+            self._restore_user_files()
+
+    def _restore_user_files(self):
+        # TODO: this mkdtemp should be in file_system?
```

Comment:
In this context is not important. We use file_system object primarily for testing purposes, so that we are able to replace it with a fake object, but this routine is so tightly coupled to the OS because of the mount command that I don't think we'll be able to write a proper unit/integration test for it anyways.